scriptmodules/emulators/jzintv.sh

```diff
@@ -61,17 +70,60 @@ function install_jzintv() {
 }
 
 function configure_jzintv() {
-    mkRomDir "intellivision"
+    local -r system="intellivision"
+    mkRomDir $system
 
-    local options=(
-        --displaysize="%XRES%x%YRES%"
-        --quiet
-        --rom-path="$biosdir"
-        --voice=1
-    )
+    local start_script="$md_inst/jzintv_launcher.sh"
+    cat > "$start_script" << _EOF_
+#! /usr/bin/env bash
+
+# \$1: width of display
+# \$2: height of display
+# \$3: --ecs=1, optional
+# \$4,5,6...: more optional parameters
+# last parameter: %ROM%
+
+jzintv_bin="$md_inst/bin/jzintv"
+
+disp_w=\$1; shift
+disp_h=\$1; shift
+
+min=\$(( \$disp_w < \$disp_h ? \$disp_w : \$disp_h ))
+if [[ \$min = \$disp_h ]] ; then
+    intv_w=\$(echo \$min | awk '{ printf "%0.f\n", \$0 * 1.3333 + .5 }')
+    intv_h=\$min
+else
+    # 90/270 rotated displays
+    intv_w=\$min
+    intv_h=\$(echo \$min | awk '{ printf "%0.f\n", \$0 / 1.3333 }')
+fi
+
+# set --gfx-verbose instead of --quiet for verbose output
+options=(
+    -f1 # fullscreen
+    --quiet
+#    --gfx-verbose
+    --displaysize="\${intv_w}x\${intv_h}"
+    --rom-path="$biosdir"
+    --voice=1
+)
+
+# ingest additional options
+while [[ \$# -gt 1 ]] ; do
+    options+=(\$1); shift
+done
```

Comment:
if you have shifted the input params why just do something like `options+=("$@")`

Reply:
The jzintv options are "space-free", while the ROM (last parameter) could contain spaces. The last parameter should be quoted to avoid shell interpretation.

Reply:
I don't see how that wouldn't work with my line above - please can you give an example ?